### PR TITLE
16# ADMIN - Wygaszanie pacjentów i lekarzy

### PR DESCRIPTION
**Pull request by MacWar**
- Add lock and unlock users accounts feature via admin panel

### CODE REVIEW

File: MedSystem/Backend/MedSystem/Controllers/AccountController.cs [hunk 1]
```diff
@@ -18,7 +20,6 @@ public class AccountController : Controller
     {
         private readonly IAccountRepository _accountRepository;
         private readonly IAuthenticationService _accountService;
-
```
Comment (Hoodster) [inline MedSystem/Backend/MedSystem/Controllers/AccountController.cs:21]: Bring back spacing here

File: MedSystem/Backend/MedSystem/Controllers/AccountController.cs [hunk 2]
```diff
@@ -53,16 +54,41 @@ public async Task<IActionResult> SignInAccount([FromBody] UserCredentialsDto cre
...
+        [Route("unlockUser")]
+        [HttpPost]
+        [Authorize(AuthenticationSchemes = JwtBearerDefaults.AuthenticationScheme, Roles = "Administrator")]
+        public async Task<IActionResult> UnlockUserAsync([FromQuery] string email)
+        {
+            await _accountRepository.UnlockUserAsync(email);
+            return Ok();
+        }
+
+
```
Comment (Hoodster) [inline MedSystem/Backend/MedSystem/Controllers/AccountController.cs:83]: Remove empty line

File: MedSystem/Backend/MedSystem/Controllers/AccountController.cs [hunk 3]
```diff
@@ -53,16 +54,41 @@ public async Task<IActionResult> SignInAccount([FromBody] UserCredentialsDto cre
...
+        }
+
+       
+        [Route("getUserByMail")]
+        [HttpGet]
+        public async Task<IActionResult> GetUserByMailAsync([FromQuery] string email)
+        {          
+            return Ok(await _accountRepository.GetUserByMailAsync(email));
+        }
+
```
Comment (Hoodster) [inline MedSystem/Backend/MedSystem/Controllers/AccountController.cs:90]: Remove empty line

File: MedSystem/Backend/MedSystem/Core/Repositories/AccountRepository/AccountRepository.cs [hunk 4]
```diff
@@ -132,6 +132,29 @@ private async Task AddToRoleAsync(User user, string role)
...
+            var endDate = new DateTime(9999, 12, 12);
+            var user = await _userManager.FindByEmailAsync(email);
+            user.LockoutEnabled = true;
+            user.LockoutEnd = endDate;
+            await _userManager.UpdateAsync(user);
+        }
+
+        public async Task UnlockUserAsync(string email)
+        {
+            var user = await _userManager.FindByEmailAsync(email);
```
Comment (Hoodster) [inline MedSystem/Backend/MedSystem/Core/Repositories/AccountRepository/AccountRepository.cs:148]: For what declaring ‘LockoutEnabled’ twice to be true? Is it false in any case?

Comment (MacWar) [reply MedSystem/Backend/MedSystem/Core/Repositories/AccountRepository/AccountRepository.cs:148]: no, it should never be false, i will remove second locoutEnabled declaration

**Pull request merged**